Objects/dictobject.c

```diff
@@ -3327,9 +3328,25 @@ dictiter_new(PyDictObject *dict, PyTypeObject *itertype)
     Py_INCREF(dict);
     di->di_dict = dict;
     di->di_used = dict->ma_used;
-    di->di_pos = 0;
     di->len = dict->ma_used;
-    if (itertype == &PyDictIterItem_Type) {
+    if (itertype == &PyDictRevIterKey_Type ||
+        itertype == &PyDictRevIterItem_Type ||
+        itertype == &PyDictRevIterValue_Type) {
+        if (dict->ma_used) {
+            di->di_pos = dict->ma_used - 1;
```

Comment:
This seem wrong.

```
d = {'a':1, 'b':2, 'c':3}
del d['b']
```

ma_used is 2, and dk_nentries is 3.  dk_nentries-1 should be used.